android/PartialSync/app/src/main/java/io/realm/todo/ProjectsActivity.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2018 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.todo;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.EditText;
+
+import java.util.Date;
+
+import io.realm.Realm;
+import io.realm.RealmResults;
+import io.realm.Sort;
+import io.realm.SyncUser;
+import io.realm.todo.model.Project;
+import io.realm.todo.ui.ProjectsRecyclerAdapter;
+
+public class ProjectsActivity extends AppCompatActivity {
+    private Realm mRealm;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_items);
+
+        setSupportActionBar(findViewById(R.id.toolbar));
+
+        findViewById(R.id.fab).setOnClickListener(view -> {
+            View dialogView = LayoutInflater.from(this).inflate(R.layout.dialog_task, null);
+            EditText taskText = dialogView.findViewById(R.id.task);
+            new AlertDialog.Builder(ProjectsActivity.this)
+                    .setTitle("Add a new project")
+                    .setView(dialogView)
+                    .setPositiveButton("Add", (dialog, which) -> mRealm.executeTransactionAsync(realm -> {
+                        Project project = new Project();
+                        String userId = SyncUser.currentUser().getIdentity();
+                        String name = taskText.getText().toString();
+
+                        // To avoid collision we use an id composed
+                        // of the name + SyncUser#id
+                        project.setId(userId + name);
```

Comment:
Why not just `UUID.getRandom().toString()`?

Reply:
it's like a composite primary key, knowing the user Id & name I can deduct the PK. but using `UUID.getRandom().toString()` is also good in this scenario